org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java

```diff
@@ -289,6 +300,21 @@ public boolean isCanceled() {
 				}
 			}
 		}
+		List<CompletionItem> tempProposals = proposals.stream().filter(prop -> prop.getKind() == CompletionItemKind.Keyword || prop.getKind() == CompletionItemKind.Snippet).collect(Collectors.toList());
```

Comment:
I would probably add a comment here explaining what we're trying to do. Something like :

"When at least one snippet has the same label as a keyword, we raise the relevance of all matching snippets above that keyword."